rego/rego.go

```diff
@@ -1044,6 +1052,10 @@ func New(options ...func(r *Rego)) *Rego {
 			WithBuiltins(r.builtinDecls)
 	}
 
+	if r.schemaSet != nil {
+		r.compiler.WithSchemas(r.schemaSet)
```

Comment:
Move the `WithSchemas` call up to L1050 where the compiler is constructed. The reason is that if users pass their own compiler, we don't want to modify the schemas here. In other words, it's up to the caller that creates to the compiler to set the schemas to use for compilation.

Reply:
Done (commit https://github.com/open-policy-agent/opa/pull/3060/commits/31f947dbded99ceb8b83440a83323a544275d71a)